src/helpers.rs

```diff
@@ -54,29 +54,57 @@ pub(crate) fn clone_and_degit(url: &str, target: &Path) -> Result<()> {
 /// For a template it should be `<template>/pallets/`
 /// For no path, it should just place it in the current working directory
 pub(crate) fn resolve_pallet_path(path: Option<String>) -> PathBuf {
-	use std::process;
-
 	if let Some(path) = path {
 		return Path::new(&path).to_path_buf();
 	}
-	// Check if inside a template
+	// Check if inside a project
 	let cwd = current_dir().expect("current dir is inaccessible");
+	match workspace_dir(None) {
+		// No workspace available, return cwd.
+		None => cwd,
+		Some(dir) => {
```

Comment:
Suggest adding checks here to ensure that the cargo workspace belongs to a parachain project.

Reply:
This sounds like the logical next step :+1: . As the description says it just naively looks for the workspace root and creates any folder there.